kernel/logger_aarch64/lib.rs

```diff
@@ -1,18 +1,19 @@
 #![no_std]
 
-use pl011_qemu::{PL011, UART1};
+use pl011_qemu::PL011;
 use log::{Record, Metadata, Log, set_logger, set_max_level, STATIC_MAX_LEVEL};
 use core::fmt::Write;
+use core::ops::DerefMut;
 use irq_safety::MutexIrqSafe;
 
-type QemuVirtUart = PL011<UART1>;
+use memory::{PhysicalAddress, MappedPages, PteFlags, get_kernel_mmi_ref, allocate_pages, allocate_frames_at};
 
 /// This wraps a UART channel handle.
-pub struct QemuVirtUartLogger {
-    pub(crate) uart: MutexIrqSafe<QemuVirtUart>,
+pub struct UartLogger {
+    pub(crate) uart: MutexIrqSafe<PL011>,
 }
```

Comment:
as with my other comment, move the `MutexIrqSafe` outside of the `UartLogger` so we don't need the unsafe `static mut`.